tests/unit/workflow/test_evaluator.py

```diff
@@ -150,12 +151,89 @@ class DictTester(base):
         class ListTester(base):
             a: List[float]
 
+
+def test__validate__metrics_test_suite__invalid_nested() -> None:
     @dataclasses.dataclass(frozen=True)
     class Nested(DataObject):
         a: int
 
     with pytest.raises(ValueError):
 
         @dataclasses.dataclass(frozen=True)
-        class NestedTester(base):
+        class NestedTester(MetricsTestSuite):
             a: Nested
+
+
+def test__validate__metrics_test_case__valid_nested() -> None:
+    @dataclasses.dataclass(frozen=True)
+    class Nested(MetricsTestCase):
+        a: int
+        b: str
+        c: float
+        d: Optional[str]
+        e: Union[int, str, float]
+
+    @dataclasses.dataclass(frozen=True)
+    class Nested2(MetricsTestCase):
+        a: int
+
+    @dataclasses.dataclass(frozen=True)
+    class NestedTester(MetricsTestCase):
+        a: int
+        b: float
+        c: List[Nested]
+        d: Optional[List[Nested]]
+        e: Union[List[Nested], List[Nested2]]
+        f: List[Union[Nested, Nested2]]
```

Comment:
What's the desired behavior for `List[Optional[Nested]]`? My assumption is that we should forbid it -- can we add a check for that?

Questions for how marina will handle this:
- How does marina know to treat the object as a `List[Nested]`? As is, my initial approach would be to check the value type, ensure that it isn't an asset or annotation, then assume that it's a nested metric if it's an Object. That's a bit messy, am I missing anything?
- How will marina handle `List[Union[Nested, Nested2]]`? Should it read all of the entries, and then use that to determine the columns to display (seems like there's no way to avoid loading all of the data, which would prevent chunking it out, but that's probably okay)?

Reply:
> How does the webapp know to treat the object as a `List[Nested]`?

No real magic or special handling required here. Test case metrics values are either scalars or lists of scalar-only objects. No need to check for things like assets or annotations.

> How will the webapp handle `List[Union[Nested, Nested2]]`?

Yeah, as suggested, union the columns. We can do this on the frontend or on the backend if chunking is required.

Reply:
> What's the desired behavior for `List[Optional[Nested]]`? My assumption is that we should forbid it

👍 Added a test an extra validation condition to disallow this.